### ISSUE TITLE
1.13 - 1.14 : @remix-run/server-runtime EntryContext is not assignable to type @remix-run/react EntryContext 🔴 

### ISSUE THREAD

**Issue by raskyer** [remix-run/remix#5615]
### What version of Remix are you using?

1.13.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

Install remix server template.

Typescript version : 4.9.5

### Expected Behavior

RemixServer doesn't complain with remixContext

### Actual Behavior

Type error with remix context.

![image](https://user-images.githubusercontent.com/16725289/222154745-65edf214-12b1-4c00-9819-77fa3030878c.png)
![image](https://user-images.githubusercontent.com/16725289/222154663-7228da00-25d2-4cfe-83e7-bbc9ca6f2556.png)

**Comment by alexhoo** [comment]
![Screenshot 2023-03-07 at 18 28 03](https://user-images.githubusercontent.com/1938383/223501496-0391d1cd-4ad4-42d0-9280-d01e85d20dfe.jpg)
Same happened to me after upgrading to 1.14

**Comment by hjonasson** [comment]
Same does not repro for me with 1.14 when I write

```TS
import { RemixServer } from "@remix-run/react";
import type { EntryContext } from "@remix-run/node";

let context: EntryContext;
export const SomeServer = () => <RemixServer url='some-url' context={context} />;
```

What is the error you are getting @raskyer ?

**Comment by alexhoo** [comment]
I should add that this happend to me since I upgraded from 1.11

**Comment by raskyer** [comment]
> Same does not repro for me with 1.14 when I write
> 
> 
> 
> ```TS
> 
> import { RemixServer } from "@remix-run/react";
> 
> import type { EntryContext } from "@remix-run/node";
> 
> 
> 
> let context: EntryContext;
> 
> export const SomeServer = () => <RemixServer url='some-url' context={context} />;
> 
> ```
> 
> 
> 
> What is the error you are getting @raskyer ?

What do you mean ? You repro or not ?

I have the exact type error in screenshot I have posted.

Upgrading to 1.14 does not solve the issue. 

I have remove my node_modules / package-lock but it doesn't solve the issue.

It's a typescript error, not node or runtime.

**Comment by alexhoo** [comment]
i have created a thread here: https://discord.com/channels/770287896669978684/1082941041306505217 
My react version is: React 17.0.2 if this helps

**Comment by raskyer** [comment]
```
Type 'import("/node_modules/@remix-run/server-runtime/dist/entry").EntryContext' is not assignable to type 'import("/node_modules/@remix-run/react/dist/entry").EntryContext'.
  Types of property 'routeModules' are incompatible.
    Type 'RouteModules<EntryRouteModule>' is not assignable to type 'RouteModules'.
      'string' index signatures are incompatible.
        Type 'EntryRouteModule' is not assignable to type 'RouteModule'.
          Types of property 'CatchBoundary' are incompatible.
            Type 'import("/node_modules/@remix-run/server-runtime/dist/routeModules").CatchBoundaryComponent | undefined' is not assignable to type 'import("/node_modules/@remix-run/react/dist/routeModules").CatchBoundaryComponent | undefined'.
              Type 'ComponentClass<{}, any>' is not assignable to type 'CatchBoundaryComponent | undefined'.
                Type 'import("/node_modules/@remix-run/server-runtime/node_modules/@types/react/index").ComponentClass<{}, any>' is not assignable to type 'React.ComponentClass<{}, any>'.
                  Types of property 'contextType' are incompatible.
                    Type 'import("/node_modules/@remix-run/server-runtime/node_modules/@types/react/index").Context<any> | undefined' is not assignable to type 'React.Context<any> | undefined'.
                      Type 'import("/node_modules/@remix-run/server-runtime/node_modules/@types/react/index").Context<any>' is not assignable to type 'React.Context<any>'.
                        Types of property 'Provider' are incompatible.
                          Type 'import("/node_modules/@remix-run/server-runtime/node_modules/@types/react/index").Provider<any>' is not assignable to type 'React.Provider<any>'.
                            Types of parameters 'props' and 'props' are incompatible.
                              Type 'React.ProviderProps<any>' is not assignable to type 'import("/node_modules/@remix-run/server-runtime/node_modules/@types/react/index").ProviderProps<any>'.
                                Types of property 'children' are incompatible.
                                  Type 'React.ReactNode' is not assignable to type 'import("/node_modules/@remix-run/server-runtime/node_modules/@types/react/index").ReactNode'.
                                    Type '{}' is not assignable to type 'ReactNode'.ts(2322)
```

**Comment by raskyer** [comment]
@alexhoo Yes Alex :) I think every one with new typescript / remix project will have it. 
It was more for future reader / reviewer, to be able to quickly jump in.

**Comment by alexhoo** [comment]
what version of React are you using @raskyer ?

**Comment by raskyer** [comment]
17.0.2

**Comment by raskyer** [comment]
Ok, it seems like @remix-run/server-runtime use @types/react at the version 18.0.15 but locally I have 17.0.53.
It seems that when both version exist within the same project it generates some weird typescript error.

One fix is to update @types/react to stick to version used by @remix-run/server-runtime.

**Comment by alexhoo** [comment]
have you tried?

**Comment by raskyer** [comment]
Now the question we should ask is :

Is it normal that `@types/react` from `@remix-run/server-runtime` is set to 18.0.15 and why it interfer with the `@types/react` a project could have. Especially because if we come from remix 1.11 the package `@types/react` is on 17.

Maybe @machour or @brophdawg11 have an idea ?

**Comment by MichaelDeBoey** [comment]
This was caused by my changes in https://github.com/remix-run/remix/pull/4801 & will be fixed once #5713 (or another solution) is merged

**Comment by brophdawg11** [comment]
This is fixed by #5713 and should be available in the next Remix release.  

We ended up removing `@types/react` from `@remix-run/server-runtime` completely since the server is React-agnostic.  This meant loosening the React-specific types exported from there (i.e., `ErrorBoundaryComponent`) to `any` and deprecating them in favor of their counterparts in `@remix-run/react`.

**Comment by MakhouT** [comment]
How exactly should this be fixed?
I am still having the issue currently. 
This is what I have in my package.json:
```
"@remix-run/css-bundle": "^2.6.0",
"@remix-run/node": "^2.6.0",
"@remix-run/react": "^2.6.0",
"@remix-run/serve": "^2.6.0",
...
"@remix-run/dev": "^2.6.0",
"@remix-run/testing": "^2.8.1",
"@types/react": "^18.2.20",
"@types/react-dom": "^18.2.7",
```

**Comment by 127** [comment]
Same error in 2.*

**Comment by brophdawg11** [comment]
Please open a new issue with a minimal reproduction and we can take a look

**Comment by 127** [comment]
I've found he error by the way. It was not a remix issue. 
The problem was at  "remix-auth-apple": "^0.1.1",
Somehow it creates conflict at ` i18next.getRouteNamespaces(**remixContext**)`
Removing this library fixed the issue
```
async function handleBrowserRequest(
  request: Request,
  responseStatusCode: number,
  responseHeaders: Headers,
  remixContext: EntryContext
) {
  const instance = createInstance();
  const lng = await i18next.getLocale(request);
  const ns = i18next.getRouteNamespaces(**remixContext**);

```
